### PR TITLE
Add ToJSON/FromJSON instances to Auto, SqlSerial

### PR DESCRIPTION
**Pull request by 3noch**

### CODE REVIEW

**Comment by 3noch** [comment]
Hang on, turns out I don't need those `toJSON` impls.

**Comment by 3noch** [comment]
Nevermind, I feel more comfortable with them there. I think leaving them out relies on the Generic instance which I don't really want.

File: beam-core/Database/Beam/Backend/SQL/Types.hs [hunk 1]
```diff
@@ -17,3 +18,8 @@ newtype SqlSerial a = SqlSerial { unSerial :: a }
   deriving (Show, Read, Eq, Ord)
 instance FromBackendRow be x => FromBackendRow be (SqlSerial x) where
   fromBackendRow = SqlSerial <$> fromBackendRow
+instance Json.FromJSON a => Json.FromJSON (SqlSerial a) where
+  parseJSON a = SqlSerial <$> Json.parseJSON a
+instance Json.ToJSON a => Json.ToJSON (SqlSerial a) where
+  toJSON (SqlSerial a) = Json.toJSON a
+  toEncoding (SqlSerial a) = Json.toEncoding a
```
Comment (tathougies) [inline beam-core/Database/Beam/Backend/SQL/Types.hs]: Can you add a newline at the end of the file here?

Comment (3noch) [reply beam-core/Database/Beam/Backend/SQL/Types.hs]: Yep will do.

**Comment by 3noch** [comment]
Done.

**Comment by 3noch** [comment]
Ready!

**Pull request merged**